slm_porosity/heat_transfer.f90

```diff
@@ -1114,6 +1140,35 @@ PURE FUNCTION laser_distribution (x_, nlocal, factor)
     laser_distribution = EXP(-SUM(factor_*(x_-x_center)**2, 2))/pi**(.5*(dim-1))
   END FUNCTION laser_distribution
 
+  FUNCTION laser_heat_source (x_, nlocal)
+    IMPLICIT NONE
+    REAL(pr), INTENT(IN) :: x_(nlocal,dim)
+    INTEGER,  INTENT(IN) :: nlocal
+    REAL(pr) :: laser_heat_source(nlocal)
+    REAL(pr) :: position(3), halfwidth(3)
+    INTEGER  :: i
+
+    position = initial_laser_position
+    position(1) = position(1) - scanning_speed*t
+    halfwidth = 1.0_pr
+    halfwidth(dim) = laser_beam_depth
+    laser_heat_source = 1.0_pr
+    DO i = 1, dim
+      laser_heat_source = laser_heat_source*laser_gaussian(x_(:,i), nlocal, position(i), halfwidth(i))
+    END DO
+    laser_heat_source = laser_heat_source*absorptivity*laser_power*power_factor
```

Comment:
There is `laser_heat_flux` function for this. You violate guideline 3.

Reply:
still working on normalization